feat(doc): add ods global accessibility page with examples (#868)
ovh/design-system: packages/storybook/stories/ovhcloud-design-system/accessibility.mdx

<Meta title="OVHcloud Design System/Accessibility" />

<Heading label="Accessibility Guide" level={ 1 } />

<Heading label="Introduction" level={ 2 } />

This guide focuses on accessibility best practices specifically for developers integrating the **OVHcloud Design System**.

While our components are built with accessibility in mind, using semantic HTML elements by default, and including the necessary ARIA attributes where needed, certain ARIA attributes, roles, and behaviors must be implemented at the integration level to ensure full accessibility in different use cases.

This means that in most cases, you don’t need to add extra ARIA attributes.

However, depending on the context, you might need to enhance a component's accessibility. This guide will help you understand:

* When and how to use ARIA properly
* Common mistakes and best practices
* How screen readers interpret different ARIA attributes

For broader accessibility information beyond the **OVHcloud Design System**, consult our <ExternalLink href="https://www.ovhcloud.com/fr/accessibility/">Accessibility Statement</ExternalLink> available on our official website.

> <Heading label="Before using ARIA" level={ 3 } />

Reviewer: heading 2? Or it's a child of Introduction on purpose?

<Heading label="Consult the Accessibility section for each component" level={ 4 } />

Each component in **OVHcloud Design System** comes with a dedicated "Accessibility" section that provides best practices and usage guidance.

<Heading label="Avoid overriding auto-accessible elements with redundant ARIA attributes" level={ 4 } />

**OVHcloud Design System** mainly uses semantic HTML elements by default that inherently support accessibility.

> Elements such as `<button>`, `<input>` have built-in roles, keyboard support, and screen reader behaviors, so they typically don't require additional ARIA attributes.

Reviewer 2: ```
<Heading label="Be careful when adding extra ARIA attributes while ODS components might already handle it" level={ 4 } />

To be the closest of what modern browsers give to users as accessible experiences, **OVHcloud Design System** aim to rely on semantic and native HTML elements as frequently if they are de facto well adapted for screen reader experiences too.
On top of what is already predefined on our components, you can find extra information regarding accessible integration in the **"Accessibility"** section of each ODS UI component.
```

* <ExternalLink href="https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#11.2">RGAA - 11.2</ExternalLink>
* <ExternalLink href="https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#11.9">RGAA - 11.9</ExternalLink>

<Heading label="aria-labelledby: Referencing another element as a label" level={ 3 } />

Use <ExternalLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby">aria-labelledby</ExternalLink> to associate an element with an existing label (instead of duplicating text).

<Canvas of={ ButtonStories.AccessibilityBadPracticeLabelledBy } sourceState="shown" />
<Icon aria-hidden="true"
      name={ ICON_NAME.circleXmark }
      style={{ color: 'var(--ods-color-critical-500)' }} /> Screen readers will announce a button.

<Canvas of={ ButtonStories.AccessibilityLabelledBy } sourceState="shown" />
<Icon aria-hidden="true"
      name={ ICON_NAME.circleCheck }
      style={{ color: 'var(--ods-color-success-500)' }} /> When the **Button** is focused, the screen reader should announce the linked label.

<Icon aria-hidden="true"
      name={ ICON_NAME.circleInfo }

>       style={{ color: 'var(--ods-color-primary-500)' }} /> Use <ExternalLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-label">aria-label</ExternalLink> when there is no visible label, and <ExternalLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby">aria-labelledby</ExternalLink> when there is one.

Reviewer: maybe use `<Message>` here instead as it is not really easy to distinguish from validation example icons above?

Reviewer: (same for other info icon message on the whole page)

import { Icon, ICON_NAME, Message, MessageBody, MessageIcon } from '@ovhcloud/ods-react';
import { Table } from '@ovhcloud/ods-react';
import { Canvas, Meta } from '@storybook/blocks';
import { ExternalLink } from '../../src/components/externalLink/ExternalLink';
import { Heading } from '../../src/components/heading/Heading';
import * as ButtonStories from '../components/button/button.stories';
import * as IconStories from '../components/icon/icon.stories';
import * as ModalStories from '../components/modal/modal.stories';
import * as SkeletonStories from '../components/skeleton/skeleton.stories';

<Meta title="OVHcloud Design System/Guides/Accessibility" />

<Heading label="Accessibility Guide" level={ 1 } />

<Heading label="Introduction" level={ 2 } />

This guide focuses on accessibility best practices specifically for developers integrating the **OVHcloud Design System**.

> While our components are built with accessibility in mind, using semantic HTML elements by default, and including the necessary ARIA attributes where needed, certain ARIA attributes, roles, and behaviors must be implemented at the integration level to ensure full accessibility in different use cases.

Reviewer 3: ```suggestion
While our components are built with accessibility in mind, using semantic HTML elements and including the necessary ARIA attributes by default. Nevertheless, ensuring full accessibility in your specific use case might require you to implement additional ARIA attributes, roles, and behaviors at the integration level.
```

However, depending on the context, you might need to enhance a component's accessibility. This guide will help you understand:

* When and how to use ARIA properly
* Common mistakes and best practices
* How screen readers interpret different ARIA attributes

For broader accessibility information beyond the **OVHcloud Design System**, consult our <ExternalLink href="https://www.ovhcloud.com/fr/accessibility/">Accessibility Statement</ExternalLink> available on our official website.

<Heading label="Before using ARIA" level={ 3 } />

<Heading label="Consult the Accessibility section for each component" level={ 4 } />

Each component in **OVHcloud Design System** comes with a dedicated **"Accessibility"** section that provides best practices and usage guidance.

> <Heading label="Be careful when adding extra ARIA attributes while ODS components might already handle it" level={ 4 } />

Reviewer 3: ```suggestion
<Heading label="Be careful when adding extra ARIA attributes as ODS components may already handle it" level={ 4 } />
```

<Heading label="radiogroup and radio roles: Making custom radio controls accessible" level={ 3 } />

<ExternalLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/radiogroup_role">radiogroup</ExternalLink> and <ExternalLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/radio_role">radio</ExternalLink> roles ensure that screen readers correctly interpret and announce the selection.

<Canvas of={ IconStories.AccessibilityBadPracticeRating } sourceState="shown" />
<Icon aria-hidden="true"
      name={ ICON_NAME.circleXmark }
      style={{ color: 'var(--ods-color-critical-500)' }} /> Screen readers will read each star individually with no context of selection group.

<Canvas of={ IconStories.AccessibilityRating } sourceState="shown" />
<Icon aria-hidden="true"
      name={ ICON_NAME.circleCheck }
      style={{ color: 'var(--ods-color-success-500)' }} /> Screen readers will announce a radio button, its label and aria-label, which item is focused on out of the total amount of items and that it is a radio group.

<Message dismissible={ false }>
  <MessageIcon name="circle-info" />

  <MessageBody>

>     Please note you should navigate in a radio group using `arrow` keys.

Reviewer 3: ```suggestion
    Please note that you should navigate within a radio group using `arrow` keys.
```
